omc3/plotting/plot_bbq.py

```diff
@@ -0,0 +1,263 @@
+"""
+Plot BBQ
+------------------------------------
+
+Provides the plotting function for the extracted and cleaned BBQ data from timber.
+
+**Arguments:**
+
+*--Required--*
+
+- **input**: BBQ data as data frame or tfs file.
+
+  Flags: **--in**
+
+*--Optional--*
+
+- **interval** *(float)*: x_axis interval that was used in calculations.
+
+  Flags: **--interval**
+- **kick**: Kick file as data frame or tfs file.
+
+- **output** *(str)*: Save figure to this location.
+
+  Flags: **--out**
+- **show**: Show plot.
+
+  Action: ``store_true``
+- **two_plots**: Plot two axis into the figure.
+
+  Flags: **--two**
+  Action: ``store_true``
+- **x_lim** *(float)*: X-Axis limits. (yyyy-mm-dd HH:mm:ss.mmm)
+
+- **y_lim** *(float)*: Y-Axis limits.
+
+
+"""
+from collections import OrderedDict
+from contextlib import suppress
+from pathlib import Path
+
+import matplotlib.dates as mdates
+import numpy as np
+from generic_parser import entrypoint, EntryPointParameters
+from generic_parser.entrypoint_parser import save_options_to_config
+from matplotlib import pyplot as plt, gridspec
+from matplotlib.ticker import FormatStrFormatter
+
+from omc3 import amplitude_detuning_analysis as ad_ana
+from omc3.definitions import formats
+from omc3.definitions.constants import PLANES
+from omc3.plotting.utils import colors as pcolors, style as pstyle
+from omc3.tune_analysis import kick_file_modifiers as kick_mod
+from omc3.tune_analysis.constants import (get_mav_window_header, get_used_in_mav_col,
+                                          get_bbq_col, get_mav_col)
+from omc3.utils import logging_tools
+
+LOG = logging_tools.get_logger(__name__)
+
+
+def get_params():
+    params = EntryPointParameters()
+    params.add_parameter(
+        flags="--in",
+        help="BBQ data as data frame or tfs file.",
+        name="input",
+        required=True,
+    )
+    params.add_parameter(
+        help="Kick file as data frame or tfs file.",
+        name="kick",
+    )
+    params.add_parameter(
+        flags="--out",
+        help="Save figure to this location.",
+        name="output",
+        type=str,
+    )
+    params.add_parameter(
+        help="Show plot.",
+        name="show",
+        action="store_true"
+    )
+    params.add_parameter(
+        help="X-Axis limits. (yyyy-mm-dd HH:mm:ss.mmm)",
+        name="x_lim",
+        type=float,
+        nargs=2,
+    )
+    params.add_parameter(
+        help="Y-Axis limits.",
+        name="y_lim",
+        type=float,
+        nargs=2,
+    )
+    params.add_parameter(
+        flags="--interval",
+        help="x_axis interval that was used in calculations.",
+        name="interval",
+        type=float,
+        nargs=2,
+    )
+    params.add_parameter(
+        flags="--two",
+        help="Plot two axis into the figure.",
+        name="two_plots",
+        action="store_true",
+    )
+    return params
```

Comment:
Unify the parameter definition, here the flags need not to be specified...In other entrypoint the dicts are used...I would change them to this notation, as that is used mostly across the repository...